x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -306,15 +306,27 @@ private Object toJavaObjectUnsignedLongAware(Block block, int position) {
      * </p>
      */
     public final void testEvaluateBlockWithoutNulls() {
-        testEvaluateBlock(driverContext().blockFactory(), driverContext(), false);
+        assumeTrue("sometimes test data are too large to fit in the memory, just skip these cases", testCase.getExpectedWarnings() == null);
```

Comment:
The `assumeTrue` seems like a temporary workaround while we're fixing the issues, right? Maybe we should comment that too.

Reply:
> The `assumeTrue` seems like a temporary workaround while we're fixing the issues, right? Maybe we should comment that too.

assumeTrue here is to check the test case does not have a warning, similar with the testCrankyEvaluateBlock with and without nulls. 5 lines below is to fail the test if the test data is too big.

Reply:
Could you change the message to say it's about warnings then please?

Sorry it took me so long to reply. I had missed the comment.